utils/versions.py

```diff
@@ -0,0 +1,156 @@
+import json
+import os
+from datetime import datetime
+from typing import Union, Optional
+from . import storage
+
+def get_version(collection_id: Union[int, str], uri: str) -> str:
```

Comment:
Sure, since I wasn't entirely convinced I wouldn't wind up reworking the whole versioning system, I hadn't yet consistently incorporated comments. I'll address docstrings throughout momentarily.